src/librbd/cache/rwl/ImageCacheState.cc

```diff
@@ -83,6 +83,78 @@ void ImageCacheState<I>::dump(ceph::Formatter *f) const {
   ::encode_json("rwl_size", size, f);
 }
 
+static void get_jsonformat(const std::string& s, JSONFormattable *f)
+{
+  JSONParser p;
+  bool result = p.parse(s.c_str(), s.size());
+  ceph_assert(result);
+  decode_json_obj(*f, &p);
+}
+
+template <typename I>
+ImageCacheState<I>* ImageCacheState<I>::get_image_cache_state(I *image_ctx) {
+#if defined(WITH_RBD_RWL)
```

Comment:
Nit: would e be compiling this code if it wasn't enabled?

Reply:
@dillaman it would compile this code when `./do_cmake  -DWITH_RBD_RWL=ON` -> `make`. if `./do_cmake  -DWITH_RBD_RWL=OFF ` (default is OFF), it wouldn't compile this code.

Reply:
@dillaman if we set rbd_rwl_enabled=false in ceph.conf and  `./do_cmake -DWITH_RBD_RWL=ON`, it would be compiling this code here. What's your suggestions in here?

Reply:
Why is it compiling any of this code if `WITH_RBD_RWL` is not on?

Reply:
@dillaman I'll check it. It shouldn't be compiling any of this codes if `WITH_RBD_RWL `is not on.